commons/src/test/java/io/aiven/kafka/connect/common/source/task/PartitionInFilenameDistributionStrategyTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.source.task;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.kafka.common.config.ConfigException;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+final class PartitionInFilenameDistributionStrategyTest {
+
+    @Test
+    void partitionInFileNameDefaultAivenS3Sink() {
+        final SourceObjectDistributionStrategy taskDistribution = new PartitionInFilenameDistributionStrategy(2,
+                "logs-\\{\\{partition}}-\\{\\{start_offset}}");
+        assertThat(taskDistribution.isPartOfTask(1, "logs-1-00112.gz")).isTrue();
+    }
+
+    @Test
+    void partitionLocationNotSetExpectException() {
+        assertThatThrownBy(() -> new PartitionInFilenameDistributionStrategy(2, "logs-23-<partition>-<start_offset>"))
+                .isInstanceOf(ConfigException.class)
+                .hasMessage(
+                        "Partition pattern {{partition}} not found, please configure the expected source to include the partition pattern.");
+
+    }
+
+    @ParameterizedTest
```

Comment:
```suggestion
    @ParameterizedTest(name = "[{index}] Pattern: {0}, Filename: {1}")
```
makes it self descriptive

Reply:
also for above tests